core/src/main/java/io/questdb/cairo/wal/seq/TableSequencerImpl.java

```diff
@@ -247,6 +248,12 @@ public long nextStructureTxn(long expectedStructureVersion, TableMetadataChange
 
         if (!metadata.isSuspended()) {
             engine.notifyWalTxnCommitted(tableToken, txn);
+            if (change.isTableRename()) {
```

Comment:
You can move the rename notification to `SequencerMetadataService`, then you don't need to use concrete class `AlterOperation` instead of `TableMetadataChange`

Reply:
That would cause two issues:
* I wouldn't have the txn for the rename in the argument at that point.
* I would have a separate `nonDataTxnCommitted` notification for the rename.

With the current approach I have exactly one notification for the table rename.